cdk/lib/__snapshots__/prism.test.ts.snap

```diff
@@ -785,6 +816,55 @@ dpkg -i /prism/prism.deb",
       },
       "Type": "AWS::IAM::Policy",
     },
+    "WazuhSecurityGroup": Object {
+      "Properties": Object {
+        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
+        "SecurityGroupEgress": Array [
+          Object {
+            "CidrIp": "0.0.0.0/0",
+            "Description": "Wazuh event logging",
+            "FromPort": 1514,
+            "IpProtocol": "tcp",
+            "ToPort": 1514,
+          },
+          Object {
+            "CidrIp": "0.0.0.0/0",
+            "Description": "Wazuh agent registration",
+            "FromPort": 1515,
+            "IpProtocol": "tcp",
+            "ToPort": 1515,
+          },
+        ],
+        "Tags": Array [
+          Object {
+            "Key": "App",
+            "Value": "prism",
+          },
+          Object {
+            "Key": "gu:cdk:version",
+            "Value": "TEST",
+          },
+          Object {
+            "Key": "gu:repo",
+            "Value": "guardian/prism",
+          },
+          Object {
+            "Key": "Stack",
+            "Value": "deploy",
+          },
+          Object {
+            "Key": "Stage",
+            "Value": Object {
+              "Ref": "Stage",
+            },
+          },
+        ],
+        "VpcId": Object {
+          "Ref": "VpcId",
+        },
+      },
+      "Type": "AWS::EC2::SecurityGroup",
+    },
   },
 }
 `;
```

Comment:
It looks like there are quite a few changes in this file which will cause security groups to be replaced (updating logical ids and descriptions). We'll need to figure out how to bring new instances up at the right time to avoid the CFN update hanging or downtime (@akash1810 has a neat trick for this, but I'm struggling to remember it!). If you haven't already, you might want to rehearse this process in `CODE` to ensure that it goes smoothly before merging this in.

Reply:
The dance I do is:
1. Apply the CFN manually (either in the web console or with `cdk deploy`)
2. Wait until the CFN update deploy "fails" with a failure to delete an in-use security group
3. Perform an autoscaling deploy using RiffRaff. This will cycle the instances using the above SG out of service allowing the SG to be deleted
4. Wait for the CFN update to finish. You should now be in a fully working state w/no downtime.
5. Perform a full deploy with RIffRaff to ensure CD would be ok

Depending on how it's timed, you sometimes have to do 2 and 3 a couple of times.

Reply:
In this case the update can actually be applied cleanly without this dance. Although it looks like we are adding/deleting security groups, upon closer inspection I can see that we are predominantly adding/deleting resources of type `AWS::EC2::SecurityGroupIngress`/`AWS::EC2::SecurityGroupEgress`. The only `AWS::EC2::SecurityGroup` which gets replaced is attached to the LB (not the instances)/ AWS seems to be able to apply all of these changes cleanly in one update without introducing downtime.